File: mobile/lib/features/projects/domain/project_models.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'project_models.freezed.dart';
part 'project_models.g.dart';

@freezed
class ProjectCampaign with _$ProjectCampaign {
  const factory ProjectCampaign({
    required String id,
    required String workspaceId,
    required String brandId,
    required String productServiceId,
    required String name,
    required String description,
    required String campaignObjective,
    required String targetPlatform,
    required String campaignType,
    @Default('ACTIVE') String status,
    DateTime? createdAt,
  }) = _ProjectCampaign;

  factory ProjectCampaign.fromJson(Map<String, dynamic> json) => _$ProjectCampaignFromJson(json);
}

@freezed
class CreateProjectRequest with _$CreateProjectRequest {
  const factory CreateProjectRequest({
    required String productServiceId,
    required String name,
    required String description,
    required String campaignObjective,
    required String targetPlatform,
    required String campaignType,
  }) = _CreateProjectRequest;

  factory CreateProjectRequest.fromJson(Map<String, dynamic> json) => _$CreateProjectRequestFromJson(json);
}

File: frontend/src/app/features/projects/project.models.ts
export type ProjectCampaignStatus = 'ACTIVE' | 'ARCHIVED';

export interface ProjectCampaign {
  readonly id: string;
  readonly workspaceId: string;
  readonly brandId: string;
  readonly productServiceId: string;
  readonly createdByUserId: string;
  readonly name: string;
  readonly description: string | null;
  readonly campaignObjective: string | null;
  readonly targetPlatform: string | null;
  readonly campaignType: string | null;
  readonly status: ProjectCampaignStatus;
  readonly createdAt: string;
  readonly updatedAt: string;
}

interface ProjectCampaignMutationPayload {
  readonly name: string;
  readonly description: string | null;
  readonly campaignObjective: string | null;
  readonly targetPlatform: string | null;
  readonly campaignType: string | null;
}

export interface CreateProjectCampaignPayload extends ProjectCampaignMutationPayload {
  readonly productServiceId: string;
}

export interface UpdateProjectCampaignPayload extends ProjectCampaignMutationPayload {
  readonly status: ProjectCampaignStatus;
}
